packages/wallet/ui/src/bridge.jsx

```diff
@@ -0,0 +1,144 @@
+// @ts-check
+/* eslint-disable import/no-extraneous-dependencies */
+import '@endo/eventual-send/shim';
+import './lockdown.js';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+Error.stackTraceLimit = Infinity;
+
+/** ISSUE: where are these defined? */
+const BridgeProtocol = /** @type {const} */ ({
+  loaded: 'walletBridgeLoaded',
+  opened: 'walletBridgeOpened',
+});
+
+/**
+ * Install a dApp "connection" where messages posted from
+ * the dApp are forwarded to localStorage and vice versa.
+ *
+ * @param {{
+ *   addEventListener: typeof window.addEventListener,
+ *   parentPost: typeof window.postMessage,
+ *   t0: number,
+ * }} io
+ */
+const installDappConnection = ({ addEventListener, parentPost, t0 }) => {
+  /** @type { string } */
+  let origin;
+
+  /** @type {(key: string, value: string) => void} */
+  let setItem;
+  /** @type {string[]} */
+  const buffer = [];
+
+  console.debug('bridge: installDappConnection');
+
+  parentPost({ type: BridgeProtocol.loaded }, '*');
+  parentPost({ type: BridgeProtocol.opened }, '*');
```

Comment:
no distinction anymore?

Reply:
I don't know. It feels a bit magic, to me. In `wallet-bridge.html`, `opened` seems to go later in the process... analogous to when the user clicks the button... but I tried it that way without luck. Perhaps Sam's recent work addressed that?